courseraprogramming/commands/oauth2.py

```diff
@@ -0,0 +1,493 @@
+#!/usr/bin/env python
+
+# Copyright 2015 Coursera
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+Helpers for working with OAuth2 / etc.
+'''
+
+import BaseHTTPServer
+import ConfigParser
+import cPickle
+import logging
+import requests
+import io
+import os
+import subprocess
+import sys
+import time
+import urlparse
+import uuid
+from sys import platform as _platform
+
+
+class ExpiredToken(Exception):
+    def __init__(self, oauth2Auth):
+        self.oauth2Auth = oauth2Auth
+
+    def __str__(self):
+        return 'Attempt to use expired token. Expired at: %s.' % \
+            self.oauth2Auth.expires
+
+
+class OAuth2Exception(Exception):
+    def __init__(self, msg):
+        self.msg = msg
+
+    def __str__(self):
+        return 'OAuth2 Protocol Exception: %(msg)s' % {
```

Comment:
you're inconsistent in your use of named parameters / parameters by order in string interpolation.